### PR TITLE
Data transfer file for logistic provider

### PR DESCRIPTION
**Pull request by riadmen**
https://github.com/metasfresh/metasfresh/issues/16727

### CODE REVIEW

File: backend/de.metas.adempiere.adempiere/base/src/main/java-gen/org/compiere/model/I_AD_Process.java [hunk 1]
```diff
@@ -1030,4 +1030,51 @@ public interface I_AD_Process
...
+	 * <br>Mandatory: false
+	 * <br>Virtual Column: false
+	 */
+	@Nullable java.lang.String getStoreProcessResultFileOn();
+
+	ModelColumn<I_AD_Process, Object> COLUMN_StoreProcessResultFileOn = new ModelColumn<>(I_AD_Process.class, "StoreProcessResultFileOn", null);
+	String COLUMNNAME_StoreProcessResultFileOn = "StoreProcessResultFileOn";
+
+	// FIXME make sure they get generated!
+	String STOREPROCESSRESULTFILEON_Server = "S";
```
Comment (teosarca) [inline backend/de.metas.adempiere.adempiere/base/src/main/java-gen/org/compiere/model/I_AD_Process.java:1056]: TODO this shall be generated!

File: backend/de.metas.adempiere.adempiere/base/src/main/java-gen/org/compiere/model/X_AD_Process.java [hunk 2]
```diff
@@ -632,4 +632,34 @@ public boolean isLogWarning()
...
+	}
+
+	@Override
+	public java.lang.String getStoreProcessResultFileOn()
+	{
+		return get_ValueAsString(COLUMNNAME_StoreProcessResultFileOn);
+	}
+
+	// FIXME make sure they get generated!
+	public static final String STOREPROCESSRESULTFILEON_Server = "S";
```
Comment (teosarca) [inline backend/de.metas.adempiere.adempiere/base/src/main/java-gen/org/compiere/model/X_AD_Process.java:648]: pls make sure you generate the X_AD_Process class

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/report/ReportResultData.java [hunk 3]
```diff
@@ -118,6 +120,23 @@ public File writeToTemporaryFile(@NonNull final String filenamePrefix)
 		}
 	}
 
+	public Path writeToDirectory(@NonNull final Path directory)
+	{
+		final String ext = MimeType.getExtensionByType(reportContentType);
+		final String fileName = "Report_" + System.currentTimeMillis() + ext;
+		final Path file = directory.resolve(fileName);
```
Comment (teosarca) [inline backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/report/ReportResultData.java:127]: what's the reason for ignoring reportFilename?

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/process/JavaProcess.java [hunk 4]
```diff
@@ -336,7 +338,7 @@ private void forwardReportDataToRecipients(final ProcessInfo pi)
 		if (reportResultDataTarget.isSaveToServerDirectory())
 		{
 			final Path targetFile = reportData.writeToDirectory(reportResultDataTarget.getServerTargetDirectoryNotNull());
-			Loggables.addLog("Saved report file to {}", targetFile);
+			getResult().addLog(getProcessInfo().getPinstanceId(), null, new BigDecimal(pi.getAdProcessId().getRepoId()), String.valueOf(targetFile));
```
Comment (teosarca) [inline backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/process/JavaProcess.java:341]: why this change?

Comment (teosarca) [reply backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/process/JavaProcess.java:341]: and why not simply calling de.metas.process.JavaProcess#addLog(java.lang.String, java.lang.Object...) ?

File: backend/de.metas.adempiere.adempiere/base/src/main/java-gen/org/compiere/model/I_AD_Process.java (hunk 1 above)
Comment (riadmen) [reply backend/de.metas.adempiere.adempiere/base/src/main/java-gen/org/compiere/model/I_AD_Process.java]: done

File: backend/de.metas.adempiere.adempiere/base/src/main/java-gen/org/compiere/model/X_AD_Process.java (hunk 2 above)
Comment (riadmen) [reply backend/de.metas.adempiere.adempiere/base/src/main/java-gen/org/compiere/model/X_AD_Process.java:653]: done

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/report/ReportResultData.java (hunk 3 above)
Comment (riadmen) [reply backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/report/ReportResultData.java:127]: when i tested on local i found that all the reportFilename is allways `report.pdf`, so one report overwrites the others and we end up we just one file (the last one)

File: backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/process/JavaProcess.java (hunk 4 above)
Comment (riadmen) [reply backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/process/JavaProcess.java]: Code updated to:
```JAVA
addLog(getProcessInfo().getPinstanceId().getRepoId(), null, new BigDecimal(pi.getAdProcessId().getRepoId()), String.valueOf(targetFile));
```

So we can filters for exports via `AD_Process_ID`:
```SQL
SELECT *
FROM AD_PInstance_Log  WHERE p_number = 500008 -- AD_Process_ID
;
```

image

**Review by teosarca — APPROVED** [review]
